pymc_marketing/mmm/mmm.py

```diff
@@ -1613,6 +1614,49 @@ def format_recovered_transformation_parameters(
 
         return channels_info
 
+    def _format_parameters_for_budget_allocator(
+        self,
+    ) -> dict[str, Any]:
+        """Format the parameters for the budget allocator.
+
+        Returns
+        -------
+        dict
+            A dictionary containing the saturation and adstock parameters.
+
+        Examples
+        --------
+        >>> self._format_parameters_for_budget_allocator()
+        >>> Output:
+        {
+            'saturation_params': {
+                'lam': array([...]),
+                'beta': array([...])
+            },
+            'adstock_params': {
+                'alpha': array([...])
+            },
+            'channels': ['x1', 'x2']
+        }
+        """
+        if self.idata is not None:
+            saturation_params: dict[str, np.ndarray] = {
+                key: self.idata.posterior[f"saturation_{key}"].values
+                for key in self.saturation.default_priors.keys()
+            }
+            adstock_params: dict[str, np.ndarray] = {
+                key: self.idata.posterior[f"adstock_{key}"].values
+                for key in self.adstock.default_priors.keys()
+            }
+        else:
+            raise ValueError("idata is not initialized.")
```

Comment:
Use `self.fit_results` here instead. It already catches and raises if the model is not fit